Tests/OpenCombineTests/Helpers/TrackingSubscriber.swift

```diff
@@ -87,6 +87,12 @@ final class TrackingSubscriberBase<Value: Equatable,
     private let _receiveCompletion: ((Subscribers.Completion<Failure>) -> Void)?
     private let _onDeinit: (() -> Void)?
 
+    var onSubscribe: ((Subscription) -> Void)?
```

Comment:
These are some hooks to make it easier to observe changes on a tracking subscriber.  I'm mostly using the `onSubscribe` in the TestHelper init since self can't be captured until after init

Reply:
maybe it's better make `_receiveSubscription` internal instead of adding a new property?

Reply:
I've intentionally left the value which returns a demand as a separate property.  In other projects I've worked on, it's helpful to have something which can inspect what's going on without having to return the correct values.  I think having the closure based observers makes it easier to write more targeted tests against specific behaviors without having to worry about changing the preconditions that are set when the underlying TrackingSubscriber is initialized.  What do you think?